sample_project/Assets/SampleViewer/Samples/BuildingFilter/BuildingFilter.cs

```diff
@@ -0,0 +1,502 @@
+                break;
+        }
+    }
+
+    private String GetLoadStatus()
+    {
+        if (buildingSceneLayer == null)
+        {
+            return "Failed";
+        }
+        var loadStatus = buildingSceneLayer.LoadStatus;
+
+        if (loadStatus == ArcGISLoadStatus.Loaded)
+        {
+            return "Loaded";
+        }
+        else if (loadStatus == ArcGISLoadStatus.Loading)
+        {
+            return "Loading";
+        }
+        else
+        {
+            var index = arcGISMapComponent.Map.Layers.GetSize() - 1;
+            arcGISMapComponent.Map.Layers.Remove(index);
+            --index;
+            buildingSceneLayer = arcGISMapComponent.Map.Layers.At(index) as ArcGISBuildingSceneLayer;
+            return "Failed";
+        }
+    }
+
+    private void AddDisciplineCategoryData()
+    {
+        DisciplineCategoryData.Clear();
+        if (buildingSceneLayer == null) return;
+
+        var firstLayers = buildingSceneLayer.Sublayers;
+        for (ulong i = 0; i < firstLayers.GetSize(); i++)
+        {
+            if (firstLayers.At(i).Name != "Full Model") continue;
+
+            var secondLayers = firstLayers.At(i).Sublayers;
+            for (ulong j = 0; j < secondLayers.GetSize(); j++)
+            {
+                secondLayers.At(j).IsVisible = true;
+                Discipline newDiscipline = new Discipline
+                {
+                    Name = secondLayers.At(j).Name
+                };
+
+                var thirdLayers = secondLayers.At(j).Sublayers;
+                for (ulong k = 0; k < thirdLayers.GetSize(); k++)
+                {
+                    thirdLayers.At(k).IsVisible = true;
+                    Category subCategory = new Category
+                    {
+                        Name = thirdLayers.At(k).Name
+                    };
+                    newDiscipline.Categories.Add(subCategory);
+                }
+                DisciplineCategoryData.Add(newDiscipline);
+            }
+        }
+
+        // Define the order
+        Dictionary<string, int> DisciplineOrder = new Dictionary<string, int>
+    {
+        { "Architectural", 0 },
+        { "Structural", 1 },
+        { "Mechanical", 2 },
+        { "Electrical", 3 },
+        { "Piping", 4 }
+    };
+
+        // Order the disciplines by the predefined order
+        DisciplineCategoryData = DisciplineCategoryData
+            .OrderBy(d => DisciplineOrder.ContainsKey(d.Name) ? DisciplineOrder[d.Name] : int.MaxValue)
+            .ToList();
+    }
+
+    private void GetStatistics()
+    {
+        if (!buildingSceneLayer) { return; }
+        var data = buildingSceneLayer.FetchStatisticsAsync();
+        data.Wait();
+        var stats = data.Get();
+        stats.TryGetValue("BldgLevel", out var count);
+        var bldgLevelMostFrequentValuesCollection = count.MostFrequentValues;
+        List<int> bldgLevelValues = new List<int>();
```

Comment:
```suggestion
        var bldgLevelValues = new List<int>();
```
var helps avoid repetition in these cases.